doc/data-structures.rst

```diff
@@ -223,18 +232,16 @@ Creating a Dataset
 ~~~~~~~~~~~~~~~~~~
 
 To make an :py:class:`~xray.Dataset` from scratch, supply dictionaries for any
-variables, coordinates and attributes you would like to insert into the
-dataset.
+variables (``data_vars``), coordinates (``coords``) and attributes (``attrs``).
 
-For the ``data_vars`` and ``coords`` arguments, keys should be the name of the
-variable and values should be scalars, 1d arrays or tuples of the form
-``(dims, data[, attrs])`` sufficient to label each array:
+``data_vars`` are supplied as a dictionary with each key as the name of the variable and each
+value as one of:
+- A :py:class:`~xray.DataArray`
+- A tuple of the form ``(dims, data[, attrs])``
+- A pandas object
 
-- ``dims`` should be a sequence of strings.
-- ``data`` should be a numpy.ndarray (or array-like object) that has a
-  dimensionality equal to the length of ``dims``.
-- ``attrs`` is an arbitrary Python dictionary for storing metadata associated
-  with a particular array.
+``coords`` are supplied as dictionary of ``{coord_name: coord}`` where the values are scalar values,
```

Comment:
It would indeed be worth formalizing what it means to be "DataArray-like". The trouble is that it depends a bit on context:

For `coords` or `data_vars`, you can supply:
- An `xray.DataArray` or `xray.Variable`
- A tuple of the form (dims, data[, attrs])
- A pandas object
- 1D numpy arrays, which are assumed to be along the given dimension
- Scalars

We don't automatically labeled dimensions (except for 1D arrays), because that's probably user error rather than what they would like to see.

For casting with the `DataArray` constructor, you can use:
- An `xray.DataArray` or `xray.Variable`
- A pandas object
- Scalars or NumPy arrays, whose dimensions will labeled `dim0`, `dim1`, etc.

Here, we don't accept tuples `(dims, data[, attrs])` because there's another, more explicit place for such arguments in the `DataArray` constructor.